File: 896.Monotonic-Array.py
# https://leetcode.com/problems/monotonic-array/description/
#
# algorithms
# Easy (58.8%)
# Total Accepted:    4.9k
# Total Submissions: 8.3k
# beats 75.07% of python submissions


class Solution(object):
    def isMonotonic(self, A):
        """
        :type A: List[int]
        :rtype: bool
        """
        length = len(A)
        increase = None

        for i in xrange(1, length):
            if A[i] == A[i - 1]:
                pass
            elif increase is None:
                increase = True if A[i] > A[i - 1] else False
            else:
                if (A[i] > A[i - 1] and not increase) or \
                        (A[i] < A[i - 1] and increase):
                    return False

        return True

File: 896.Monotonic-Array.java
// https://leetcode.com/problems/monotonic-array/
//
// algorithms
// Easy (56.09%)
// Total Accepted:    61,928
// Total Submissions: 110,418


class Solution {
    public boolean isMonotonic(int[] A) {
        int isInc = 0;
        int length = A.length;

        for (int i = 1; i < length; i++) {
            if (A[i] == A[i - 1]) {
                continue;
            } else if (isInc == 0) {
                isInc = A[i] > A[i - 1] ? 1 : -1;
            } else {
                if (isInc == 1 && A[i] < A[i - 1]) {
                    return false;
                }
                if (isInc == -1 && A[i] > A[i - 1]) {
                    return false;
                }
            }
        }

        return true;
    }
}